lib/formElement/Date.js

```diff
@@ -0,0 +1,27 @@
+( function ( mw, $, undefined ) {
+	mw.ext.forms.formElement.Date = function() {};
+
+	OO.inheritClass( mw.ext.forms.formElement.Date, mw.ext.forms.formElement.InputFormElement );
+
+	mw.ext.forms.formElement.Date.prototype.getElementConfig = function() {
+		var config = mw.ext.forms.formElement.Date.parent.prototype.getElementConfigInternal.call( this );
+
+		config.value.type = 'date';
+		config.value.label =  mw.message( 'mwstake-formengine-label-date-default' ).text();
+		return this.returnConfig( config );
+	};
+
+	mw.ext.forms.formElement.Date.prototype.getType = function() {
+		return "date";
+	};
+
+	mw.ext.forms.formElement.Date.prototype.getWidgets = function() {
+		return {
+			view: mw.ext.forms.widget.view.TextView,
+			edit: mw.ext.forms.widget.edit.Date
```

Comment:
If you are not adding anything is the edit widget, no need to have it, you can just use `mw.widgets.DateInputWidget `directly

Reply:
done